com.github.vladimiry.ElectronMail.json

```diff
@@ -0,0 +1,88 @@
+{
+  "app-id": "com.github.vladimiry.ElectronMail",
+  "base": "org.electronjs.Electron2.BaseApp",
+  "base-version": "19.08",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "19.08",
+  "sdk": "org.freedesktop.Sdk",
+  "branch": "beta",
+  "separate-locales": false,
+  "command": "electron-mail",
+  "rename-desktop-file": "electron-mail.desktop",
+  "rename-icon": "electron-mail",
+  "finish-args": [
+    "--device=dri",
+    "--env=HUNSPELL_DICTIONARIES=/app/ElectronMail/usr/share/hunspell/",
+    "--filesystem=xdg-download",
+    "--share=ipc",
+    "--share=network",
+    "--socket=pulseaudio",
+    "--socket=x11",
+    "--talk-name=org.freedesktop.Notifications",
+    "--talk-name=org.freedesktop.secrets",
+    "--talk-name=org.kde.StatusNotifierWatcher"
+  ],
+  "modules": [
+    "shared-modules/libsecret/libsecret.json",
+    {
+      "name": "squashfs",
+      "buildsystem": "simple",
+      "build-commands": [
+        "make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} install INSTALL_DIR=${FLATPAK_DEST}/bin"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/plougher/squashfs-tools/archive/4.4.tar.gz",
+          "sha256": "a7fa4845e9908523c38d4acf92f8a41fdfcd19def41bd5090d7ad767a6dc75c3"
+        },
+        {
+          "type": "patch",
+          "path": "xz_support.patch"
+        }
+      ]
+    },
+    {
+      "name": "electron-mail",
+      "buildsystem": "simple",
+      "build-commands": [
+        "unsquashfs -dest /app/ElectronMail/ electron-mail.snap -e usr/share/hunspell/",
+        "rm electron-mail.snap",
+        "ar xo electron-mail.deb",
+        "tar xf data.tar.xz -C /app --strip-components=2 ./opt ./usr",
+        "rm -f control.tar.gz data.tar.xz debian-binary electron-mail.deb",
+        "rm -rf /app/share/icons/hicolor/1024x1024/",
+        "install -Dm755 electron-mail-entrypoint.sh /app/bin/electron-mail",
+        "chmod -R a-s,go+rX,go-w /app/ElectronMail",
+        "desktop-file-edit --set-key=Exec --set-value=\"/app/bin/electron-mail --js-flags=\\\"--max-old-space-size=6144\\\" %U\" /app/share/applications/electron-mail.desktop",
+        "install -Dm644 -t /app/share/metainfo ${FLATPAK_ID}.metainfo.xml"
+      ],
+      "sources": [
+        {
+          "type": "file",
+          "dest-filename": "electron-mail.snap",
+          "url": "https://github.com/vladimiry/ElectronMail/releases/download/v4.6.0/electron-mail-4.6.0-linux-amd64.snap",
+          "sha256": "cf8d55c8a0eb5047288195d755006c67d9b0e56df1603c88e76392acafa1bca0"
+        },
+        {
+          "type": "file",
+          "dest-filename": "electron-mail.deb",
+          "url": "https://github.com/vladimiry/ElectronMail/releases/download/v4.6.0/electron-mail-4.6.0-linux-amd64.deb",
+          "sha256": "2cbc115bb9a1895ae6e158534feb751f46ccb9ba7dbe180a01e4685caa470488"
+        },
```

Comment:
I'm not sure why you use both the deb & the snap file?

Reply:
Snap file serves only the hunspell dictionaries. I'd prefer not to use snap as an app source since it includes extra/snap-specific stuff to cut out. If its structure changes in the future I might miss to update the cutting out rules and we get messy flatpak package then. The deb package is a plain one, it doesn't include the extra stuff, so it's better to use it as an app source.

Reply:
Why not just build hunspell dictionaries as part of this flatpak?

Reply:
This is the way it's done now. [Here](https://github.com/flathub/flathub/pull/1656/files#diff-a852616740a224c7f70477fed607c4b5R49) we unpack only the `usr/share/hunspell/` directory from the snap file to `/app/ElectronMail/` location. [Here](https://github.com/flathub/flathub/pull/1656/files#diff-a852616740a224c7f70477fed607c4b5R15) the respective env var is set so then [the app properly resolves the hunspell dictionaries dir](https://github.com/vladimiry/ElectronMail/blob/3480dd222a9bea26be2853508b7ae3209a677645/src/electron-main/spell-check/setup.ts#L55).

Reply:
Well, you  just build hunspell here without depending on the snap package for that...

Reply:
> Well, you can just build hunspell here without depending on the snap package for that...

It might be so but I'd really prefer the single source of truth practice would be applied here. The app build process includes taking the dictionaries from https://github.com/wooorm/dictionaries and then packing them into the specific package types in a certain structure. So I'd rather disable the spellchecking feature completely for the flatpak package rather than enabling the individual to flatpak package hunspell dictionaries preparing process.